ui/src/AzureUserInput.ts

```diff
@@ -0,0 +1,104 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Memento, QuickPickItem, QuickPickOptions } from 'vscode';
+import * as vscode from 'vscode';
+import { IAzureQuickPickItem, IAzureQuickPickOptions, IAzureUserInput } from '../index';
+import { DialogResponses } from './DialogResponses';
+import { UserCancelledError } from './errors';
+import { localize } from './localize';
+import { randomUtils } from './utils/randomUtils';
+
+export class AzureUserInput implements IAzureUserInput {
+    private readonly _persistence: Memento;
+
+    public constructor(persistence: Memento) {
+        this._persistence = persistence;
+    }
+
+    public async showQuickPick<T extends QuickPickItem>(items: T[] | Thenable<T[]>, options: QuickPickOptions): Promise<T> {
+        if (options.ignoreFocusOut === undefined) {
+            options.ignoreFocusOut = true;
+        }
+
+        let persistenceKey: string | undefined;
+        const hashData: string | undefined = (<IAzureQuickPickOptions>options).id || options.placeHolder;
```

Comment:
the name "hashData" is very confusing, I kept thinking it was the actual persistence data. How about "unhashedKey"?